### ISSUE TITLE
Cannot leave sync chain when offline/error

### ISSUE THREAD

**Issue by darkdh** [brave/brave-browser#19199]
## Steps to Reproduce

   1. Turn off network
   2. Create sync chain offline
   3. Try to leave sync chain

## Actual result:

 It will stuck in the same sync chain until network came online or server error resolved

## Expected result:
Reset should take effect immediately

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome?

**Comment by GeetaSarvadnya** [comment]
Verification is in-progress

```

Brave | 1.31.91 Chromium: 95.0.4638.69 (Official Build) (64-bit)
-- | --
Revision | 6a1600ed572fedecd573b6c2b90a22fe6392a410-refs/branch-heads/4638@{#984}
OS | Windows 10 Version 21H1 (Build 19043.1288)

```
- Verified the STR from the description

**`Initial Sync Chain`** | **`Disabled Network`** | **`Leaving Sync Chain`** | **`Left Sync Chain`** | **`Left Sync Chain`** 
-------------------------|---------------------------|---------------------------|------------------------|------------------------

**Comment by btlechowski** [comment]
Looks similar to https://github.com/brave/brave-browser/issues/10879

**Comment by stephendonner** [comment]
Verified `PASSED` using Brave `1.31.91, Chromium 95.0.4638.69` on Android 9, Google Pixel XL.

Steps:
1. new install
2. created Sync chain
3. added desktop machine using Sync code words
4. disconnected from Wi-Fi
5. tapped on the trash-can icon to `Delete` / `Remove` the Sync seed device (Google Pixel XL) from the Sync chain

Confirmed I was able to leave the Sync chain successfully while offline.

example | example | example | example
---------|----------|---------|----------
![Screenshot_20211109-114549](https://user-images.githubusercontent.com/387249/140995748-7bc23070-9087-43ae-9014-cddc3c2d2230.png) | ![Screenshot_20211109-114618](https://user-images.githubusercontent.com/387249/140995776-678c9629-e02e-4c7a-a599-439597a1b3ce.png) | ![Screenshot_20211109-115116](https://user-images.githubusercontent.com/387249/140995799-c5e4c3de-27ed-4ce5-8ead-eb9a9404f8eb.png) | ![Screenshot_20211109-114724](https://user-images.githubusercontent.com/387249/140995832-ccf99b0a-6d58-4001-a8ef-43b52e5ce550.png)
